### PR TITLE
librbd: update internals to use optional separate data pool

### PR DESCRIPTION
**Pull request by dillaman**

### CODE REVIEW

**Comment by dillaman** [comment]
This PR is rebased from PR #11355

File: src/test/librbd/test_librbd.cc [hunk 1]
```diff
@@ -1420,6 +1420,97 @@ TEST_F(TestLibRBD, TestIOWithIOHint)
...
+  REQUIRE_FORMAT_V2();
+
+  rados_ioctx_t ioctx;
+  rados_ioctx_create(_cluster, m_pool_name.c_str(), &ioctx);
+
+  std::string data_pool_name = create_pool(true);
+  std::string orig_data_pool_name;
+  ASSERT_EQ(0, _rados.conf_get("rbd_default_data_pool", orig_data_pool_name));
+  ASSERT_EQ(0, _rados.conf_set("rbd_default_data_pool",
+                               data_pool_name.c_str()));
```
Comment (trociny) [inline src/test/librbd/test_librbd.cc]: @dillaman Unfortunately, we are still not allowed to change string options on the fly.
```
% RBD_FEATURES=109 ./bin/ceph_test_librbd --gtest_filter=TestLibRBD.TestDataPoolIO            
Note: Google Test filter = TestLibRBD.TestDataPoolIO
[==========] Running 1 test from 1 test case.
[----------] Global test environment set-up.
[----------] 1 test from TestLibRBD
seed 24189
[ RUN      ] TestLibRBD.TestDataPoolIO
/home/mgolub/ceph/ceph.upstream/src/test/librbd/test_librbd.cc:1434: Failure
      Expected: 0
To be equal to: _rados.conf_set("rbd_default_data_pool", data_pool_name.c_str())
      Which is: -38
[  FAILED  ] TestLibRBD.TestDataPoolIO (2494 ms)
[----------] 1 test from TestLibRBD (2494 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test case ran. (5292 ms total)
[  PASSED  ] 0 tests.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] TestLibRBD.TestDataPoolIO

 1 FAILED TEST
```

**Comment by dillaman** [comment]
@trociny Swapped the test to use image options instead

**Comment by trociny** [comment]
lgtm

**Pull request merged**